judge/judgedaemon.main.php

```diff
@@ -16,6 +16,72 @@
 $endpoints = [];
 $domjudge_config = [];
 
+function dj_getopt(string $short_options, array $long_options = []): array
+{
+    global $argv;
+    define('GETOPT_REGEX', "/^([a-zA-Z0-9]:{0,2})*$/");
+    $options = getopt($short_options, $long_options);
+    if (preg_match(GETOPT_REGEX, $short_options) !== 1) {
+        echo "Error: short options format specified is invalid.\n";
+        usage();
+    }
+    if ($options===false || !is_array($argv)) {
+        echo "Error: parsing options failed.\nPlease check: `register_argc_arg` in php.ini.\n";
+        usage();
+    }
+    check_unknown_cli_options($options);
```

Comment:
I think this function can be simplified (and then perhaps inlined?) to something like:
```
foreach (array_slice($argv, 1) as $arg) {
  if (str_starts_with($arg, '-') && !array_key_exists(ltrim($arg, '-'), $options)) {
    echo "Error: Unknown option: $arg\n";
    usage();
  }
}
```